### PR TITLE
BOT: collector ftp(s)

### PR DESCRIPTION
**Pull request by robcza**
Bots for collecting files from FTP and FTPS, both bots include automatic unzipping

Possible issue is no proxy support

### CODE REVIEW

**Comment by sebix** [comment]
Please write a least a dummy test, so the requirements and syntax will be checked as here: https://github.com/certtools/intelmq/blob/master/intelmq/tests/bots/collectors/alienvault_otx/test_collector.py

File: intelmq/bots/collectors/ftp/collector_ftps.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+import ssl
+import zipfile
+import io
+
+from intelmq.lib.bot import Bot
+from intelmq.lib.harmonization import DateTime
+from intelmq.lib.message import Report
+
+
+# https://stackoverflow.com/questions/12164470/python-ftp-implicit-tls-connection-issue
```
Comment (sebix) [inline intelmq/bots/collectors/ftp/collector_ftps.py]: If you copied code from there, attribution is required under cc by-sa 3.0: https://blog.stackoverflow.com/2009/06/attribution-required/

File: intelmq/bots/collectors/ftp/collector_ftps.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+        for raw_report in raw_reports:
+            report = Report()
+            report.add("raw", raw_report, sanitize=True)
+            report.add("feed.name", self.parameters.feed, sanitize=True)
+            report.add("feed.url", self.parameters.ftps_host + ':' +
+                       str(self.parameters.ftps_port), sanitize=True)
+            report.add("feed.accuracy", self.parameters.accuracy, sanitize=True)
+            time_observation = DateTime().generate_datetime_now()
+            report.add('time.observation', time_observation, sanitize=True)
```
Comment (sebix) [inline intelmq/bots/collectors/ftp/collector_ftps.py]: This is done by the library: https://github.com/certtools/intelmq/blob/master/intelmq/lib/message.py#L282

**Comment by robcza** [comment]
@sebix thank you for the comments, will fix and include the attribution

File: intelmq/bots/collectors/ftp/collector_ftp.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+                raw_reports.append(zfp.read(filename))
+
+        for raw_report in raw_reports:
+            report = Report()
+            report.add("raw", raw_report, sanitize=True)
+            report.add("feed.name", self.parameters.feed, sanitize=True)
+            report.add("feed.url", self.parameters.ftp_host + ':' +
+                       str(self.parameters.ftp_port), sanitize=True)
+            report.add("feed.accuracy", self.parameters.accuracy, sanitize=True)
+            time_observation = DateTime().generate_datetime_now()
```
Comment (sebix) [inline intelmq/bots/collectors/ftp/collector_ftp.py]: this line is obsolete

File: intelmq/bots/collectors/ftp/collector_ftps.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+                raw_reports.append(zfp.read(filename))
+
+        for raw_report in raw_reports:
+            report = Report()
+            report.add("raw", raw_report, sanitize=True)
+            report.add("feed.name", self.parameters.feed, sanitize=True)
+            report.add("feed.url", self.parameters.ftps_host + ':' +
+                       str(self.parameters.ftps_port), sanitize=True)
+            report.add("feed.accuracy", self.parameters.accuracy, sanitize=True)
+            time_observation = DateTime().generate_datetime_now()
```
Comment (sebix) [inline intelmq/bots/collectors/ftp/collector_ftps.py]: this line is obsolete

File: intelmq/bots/collectors/ftp/collector_ftps.py [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+import ssl
+import zipfile
+import io
+
+from intelmq.lib.bot import Bot
+from intelmq.lib.harmonization import DateTime
+from intelmq.lib.message import Report
+
+
+# BEGIN content from Stack Overflow
```
Comment (sebix) [inline intelmq/bots/collectors/ftp/collector_ftps.py]: I'd also explicitly add `cc by-sa 3.0`

**Comment by sebix** [comment]
Is it final?

**Comment by aaronkaplan** [comment]
ping? @robcza . Unit tests?

**Pull request merged**